### PR TITLE
#139 Implemented a sample scene for displaying the area around the sel…

### PR DESCRIPTION
**Pull request by HoloMoto**
SampleProject\Assets\Microsoft.Maps.Unity.Examples, I created a CurrentLocationExample folder and created a sample scene and sample script in it.

Please refer to issue #139.

I have confirmed that it works with the actual HoloLens 2 device.

The one I had already submitted in #140  had extra commits in it, so I deleted the remote branch once to remove them, and then uploaded it again.

Also, I am willing to work on any fixes.

### CODE REVIEW

File: SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using Microsoft.Geospatial;
+
+/// <summary>
+/// This class retrieves the current location of the user via the connected network and returns the latitude and longitude to the MapRenderer.
+/// It only works on the actual device.
+/// </summary>
+public class CurrentLocation : MonoBehaviour
+{
+    public MapRenderer mapRenderer;
+
```
Comment (kircher1) [inline SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs:21]: The convention we are using is private field + [SerializeField] attribute, with underscore for the naming convention on the private fields, e.g.

    [SerializeFIeld]
    private MapRenderer _mapRenderer = null;

If you want a public accessor, use a property that is backed by the private fields... If you think code access isn't necessary though, don't worry about adding properties.

File: SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        debugText.text = "Initialization.";
+       
+#if UNITY_WSA && !UNITY_EDITOR
+        var accessStatus = await Geolocator.RequestAccessAsync();
+        switch (accessStatus)
+        {
+            case GeolocationAccessStatus.Allowed:
+                debugText.text = "Waiting for update...";
+                Geolocator geolocator = new Geolocator { DesiredAccuracyInMeters = _desireAccuracyInMetersValue };
+                Geoposition pos = await geolocator.GetGeopositionAsync();
```
Comment (kircher1) [inline SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs:36]: `position` or `geoposition`

File: SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+#if UNITY_WSA && !UNITY_EDITOR
+    private void UpdateLocationData(Geoposition position)
+    {
+        if (position == null)
+        {
+            debugText.text = "No data";
+        }
+        else
+        {
+            debugText.text = position.Coordinate.Point.Position.Latitude.ToString() + "\n" + position.Coordinate.Point.Position.Longitude.ToString();
```
Comment (kircher1) [inline SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs:61]: consider pulling `position.Coordinate.Point.Position` into a local variable.

File: SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+/// It only works on the actual device.
+/// </summary>
+public class CurrentLocation : MonoBehaviour
+{
+    public MapRenderer mapRenderer;
+    
+    public TextMeshPro debugText;
+    
+    private uint _desireAccuracyInMetersValue = 0;
+    async void Start()
```
Comment (kircher1) [inline SampleProject/Assets/Microsoft.Maps.Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs:25]: private

**Review by kircher1 — COMMENTED** [review]
I had a chance to try the scene on the HoloLens 2. One issue I ran into is that the default starting position of the map ends up behind the surface mesh generated for my surrounding space.

With the current scene setup, there's no way to replace the map so it's not usable for me.

However, there are three changes you can make which involve copying some GameObjects we use in other scenes and hooking them up accordingly.

1) Add the DisplayTypeScale component to the Map GameObject

image

2) Make sure TapToPlace component on the Map GameObject is using the correct values and events. This scene doesn't have a MixedRealityMapInteractionHandler, so those events can be ignored.

image

3)Copy over a **NearSettingsMenu** prefab from another scene, e.g. MapPinExample. Open up it's **PlaceMapButton** and **ResizeMapButton** GameObjects. Find the OnClick events under the Interactable component, and hook up the events following the same convention as the other scenes, like MapPinExample.

After this, you should have a menu floating nearby when the scene runs on a HoloLens. The menu lets you place the map somewhere else in the room, or resize it.

**Comment by HoloMoto** [comment]
@kircher1 
Thanks for the review.
Unity.Examples/CurrentLocationExample/Scripts/CurrentLocation.cs according to your rules.

I've confirmed that it works on HoloLens 2.

Thank you very much!

**Review by kircher1 — APPROVED** [review]

**Pull request merged**